### PR TITLE
Add New Python Pattern: SSH-able EMR cluster via EC2 Instance

### PR DESCRIPTION
**Pull request by tommywhelan**
PR for validated idea from existing issue that I created:
https://github.com/aws-samples/aws-cdk-examples/issues/765

The pattern creates an EMR cluster that is in a Private Subnet, which can be SSH-ed into in a secure fashion via a EC2 instance in the VPC's public subnet.

This pattern provides an accelerator for developing on EMR, by reducing the time/ # of steps in which a user can gain secure access to the cluster.

---

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license].

[Apache 2.0 license]: https://www.apache.org/licenses/LICENSE-2.0

### CODE REVIEW

File: python/ssh-into-emr-cluster/emr_pattern/emr_pattern_stack.py [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+            ),
+            security_group=ec2_security_group,
+            key_name="emr_test"
+        )
+
+        # S3 Bucket used for: logging, storing emr scripts, Spark job input data, and Spark job output data
+        s3_bucket = s3.Bucket(
+            self,
+            "s3_bucket",
+            bucket_name=f"emr-cdk-pattern-{Aws.ACCOUNT_ID}-{str(uuid4())[:5]}",
```
Comment (kaizio) [inline python/ssh-into-emr-cluster/emr_pattern/emr_pattern_stack.py:100]: This is going to cause a bucket to get created on each synth/deploy even you have no changes.

File: python/ssh-into-emr-cluster/requirements.txt [hunk 2]
```diff
@@ -0,0 +1,2 @@
+aws-cdk-lib==2.49.0
```
Comment (kaizio) [inline python/ssh-into-emr-cluster/requirements.txt:1]: update to latest

File: python/ssh-into-emr-cluster/emr_pattern/emr_pattern_stack.py (hunk 1 above)
Comment (tommywhelan) [reply python/ssh-into-emr-cluster/emr_pattern/emr_pattern_stack.py]: @kaizio Thanks for checking through code + finding this bug, good callout! I have made another commit to my fork removing this UUID section for bucket name, as it shouldnt be necessary.

File: python/ssh-into-emr-cluster/requirements.txt (hunk 2 above)
Comment (kaizio) [reply python/ssh-into-emr-cluster/requirements.txt:1]: Would you be able to update this to latest. 2.49 is pretty old released back in Oct 27, 2022.

Comment (tommywhelan) [reply python/ssh-into-emr-cluster/requirements.txt:1]: ^sure! this pattern was worked out a while back, so I guess that's why it's outdated

Let me upgrade CDK version, retest, and follow up here today

**Comment by tommywhelan** [comment]
@kaizio Fixed! 

Just tested with new CDK version and worked smoothly. From my end, this pattern is good to go

**Review by kaizio — COMMENTED** [review]
:+1:

**Pull request merged**